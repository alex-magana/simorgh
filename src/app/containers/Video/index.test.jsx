import {
  shouldShallowMatchSnapshot,
  isNull,
} from '../../helpers/tests/testHelpers';
import {
  NoData,
  NoAresMedia,
  VideoClipGlobalWithCaption,
  VideoClipGlobalWithoutCaption,
  VideoClipGlobalPortrait,
  VideoClipUkWithGuidance,
  VideoClipNonUk,
  AudioClipGlobalGuidance,
  AudioClipUk,
  AudioClipNonUk,
  AudioEpisodeGlobal,
} from './fixtureData';

describe('AudioVideo', () => {
  describe('with no data', () => {
    isNull('canonical - should render null', NoData({ platform: 'canonical' }));
    isNull('amp - should render null', NoData({ platform: 'amp' }));
  });

  describe('with no aresMedia block', () => {
    isNull(
      'canonical - should render null',
      NoAresMedia({ platform: 'canonical' }),
    );
    isNull('amp - should render null', NoAresMedia({ platform: 'amp' }));
  });

  describe('with Video data', () => {
    shouldShallowMatchSnapshot(
      'canonical - should render the video without a caption',
      VideoClipGlobalWithoutCaption({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render the video without a caption',
      VideoClipGlobalWithoutCaption({ platform: 'amp' }),
    );

    shouldShallowMatchSnapshot(
      'canonical - should render portrait video with global visibility',
      VideoClipGlobalPortrait({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render portrait video with global visibility',
      VideoClipGlobalPortrait({ platform: 'amp' }),
    );

    shouldShallowMatchSnapshot(
      'canonical - should render the video without a caption',
      VideoClipUkWithGuidance({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render the video without a caption',
      VideoClipUkWithGuidance({ platform: 'amp' }),
    );

    shouldShallowMatchSnapshot(
      'canonical - should render the video without a caption',
      VideoClipNonUk({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render the video without a caption',
      VideoClipNonUk({ platform: 'amp' }),
    );
  });

  describe('with Video data and a caption', () => {
    shouldShallowMatchSnapshot(
      'canonical - should render the video and caption',
      VideoClipGlobalWithCaption({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render the video and caption',
      VideoClipGlobalWithCaption({ platform: 'amp' }),
    );
  });

  describe('with Audio data', () => {
    shouldShallowMatchSnapshot(
      'canonical - should render audio clip global with guidance',
      AudioClipGlobalGuidance({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render audio clip global with guidance',
      AudioClipGlobalGuidance({ platform: 'amp' }),
    );

    shouldShallowMatchSnapshot(
      'canonical - should render audio clip UK only',
      AudioClipUk({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render audio clip UK only',
      AudioClipUk({ platform: 'amp' }),
    );

    shouldShallowMatchSnapshot(
      'canonical - should render audio clip non-UK only',
      AudioClipNonUk({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render audio clip non-UK only',
      AudioClipNonUk({ platform: 'amp' }),
    );

    shouldShallowMatchSnapshot(
      'canonical - should render audio episode global',
      AudioEpisodeGlobal({ platform: 'canonical' }),
    );
    shouldShallowMatchSnapshot(
      'amp - should render audio episode global',
      AudioEpisodeGlobal({ platform: 'amp' }),
    );
  });
});
