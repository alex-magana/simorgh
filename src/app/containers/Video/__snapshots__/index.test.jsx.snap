// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video with data and a caption should render the video and caption 1`] = `
<Context.Provider
  value={
    Object {
      "isUK": true,
      "origin": "http://localhost",
      "platform": undefined,
    }
  }
>
  <VideoContainer
    blocks={
      Array [
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "blockId": "urn:bbc:ares::clip:p01k6msm",
                "model": Object {
                  "advertising": true,
                  "caption": null,
                  "embedding": true,
                  "format": "audio_video",
                  "id": "p01k6msm",
                  "image": null,
                  "imageCopyright": "BBC",
                  "imageUrl": "ichef.test.bbci.co.uk/images/ic/$recipe/p01k6mtv.jpg",
                  "subType": "clip",
                  "syndication": Object {
                    "destinations": Array [],
                  },
                  "synopses": Object {
                    "short": "They may be tiny, but us humans could learn a thing or two from ants.",
                  },
                  "title": "Five things ants can teach us about management",
                  "versions": Array [
                    Object {
                      "availableTerritories": Object {
                        "nonUk": true,
                        "uk": true,
                      },
                      "availableUntil": null,
                      "duration": 191,
                      "types": Array [
                        "Original",
                      ],
                      "versionId": "p01k6msp",
                      "warnings": Object {
                        "long": "Contains strong language and adult humour.",
                        "short": "Contains strong language and adult humour.",
                      },
                    },
                  ],
                },
                "type": "aresMediaMetadata",
              },
              Object {
                "model": Object {
                  "blocks": Array [
                    Object {
                      "model": Object {
                        "copyrightHolder": "BBC",
                        "height": 1080,
                        "locator": "ichef.test.bbci.co.uk/images/ic/$recipe/p01k6mtv.jpg",
                        "originCode": null,
                        "width": 1920,
                      },
                      "type": "rawImage",
                    },
                    Object {
                      "model": Object {
                        "blocks": Array [
                          Object {
                            "model": Object {
                              "blocks": Array [
                                Object {
                                  "model": Object {
                                    "blocks": Array [
                                      Object {
                                        "model": Object {
                                          "attributes": Array [],
                                          "text": "Ants",
                                        },
                                        "type": "fragment",
                                      },
                                    ],
                                    "text": "Ants",
                                  },
                                  "type": "paragraph",
                                },
                              ],
                            },
                            "type": "text",
                          },
                        ],
                      },
                      "type": "altText",
                    },
                  ],
                },
                "type": "image",
              },
            ],
          },
          "type": "aresMedia",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "blocks": Array [
                    Object {
                      "model": Object {
                        "blocks": Array [
                          Object {
                            "model": Object {
                              "attributes": Array [],
                              "text": "p01k6msm: Video, Clip, UK and non-UK, guidance, subtitles (about bees)",
                            },
                            "type": "fragment",
                          },
                        ],
                        "text": "p01k6msm: Video, Clip, UK and non-UK, guidance, subtitles (about bees)",
                      },
                      "type": "paragraph",
                    },
                  ],
                },
                "type": "text",
              },
            ],
          },
          "type": "caption",
        },
      ]
    }
  />
</Context.Provider>
`;

exports[`Video with data and a caption should render the video without a caption 1`] = `
<Context.Provider
  value={
    Object {
      "isUK": true,
      "origin": "http://localhost",
      "platform": undefined,
    }
  }
>
  <VideoContainer
    blocks={
      Array [
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "blockId": "urn:bbc:ares::clip:p01k6msm",
                "model": Object {
                  "advertising": true,
                  "caption": null,
                  "embedding": true,
                  "format": "audio_video",
                  "id": "p01k6msm",
                  "image": null,
                  "imageCopyright": "BBC",
                  "imageUrl": "ichef.test.bbci.co.uk/images/ic/$recipe/p01k6mtv.jpg",
                  "subType": "clip",
                  "syndication": Object {
                    "destinations": Array [],
                  },
                  "synopses": Object {
                    "short": "They may be tiny, but us humans could learn a thing or two from ants.",
                  },
                  "title": "Five things ants can teach us about management",
                  "versions": Array [
                    Object {
                      "availableTerritories": Object {
                        "nonUk": true,
                        "uk": true,
                      },
                      "availableUntil": null,
                      "duration": 191,
                      "types": Array [
                        "Original",
                      ],
                      "versionId": "p01k6msp",
                      "warnings": Object {
                        "long": "Contains strong language and adult humour.",
                        "short": "Contains strong language and adult humour.",
                      },
                    },
                  ],
                },
                "type": "aresMediaMetadata",
              },
              Object {
                "model": Object {
                  "blocks": Array [
                    Object {
                      "model": Object {
                        "copyrightHolder": "BBC",
                        "height": 1080,
                        "locator": "ichef.test.bbci.co.uk/images/ic/$recipe/p01k6mtv.jpg",
                        "originCode": null,
                        "width": 1920,
                      },
                      "type": "rawImage",
                    },
                    Object {
                      "model": Object {
                        "blocks": Array [
                          Object {
                            "model": Object {
                              "blocks": Array [
                                Object {
                                  "model": Object {
                                    "blocks": Array [
                                      Object {
                                        "model": Object {
                                          "attributes": Array [],
                                          "text": "Ants",
                                        },
                                        "type": "fragment",
                                      },
                                    ],
                                    "text": "Ants",
                                  },
                                  "type": "paragraph",
                                },
                              ],
                            },
                            "type": "text",
                          },
                        ],
                      },
                      "type": "altText",
                    },
                  ],
                },
                "type": "image",
              },
            ],
          },
          "type": "aresMedia",
        },
      ]
    }
  />
</Context.Provider>
`;

exports[`Video with data should return a valid VideoContainer 1`] = `
<React.Fragment>
  <ForwardRef>
    <Video
      holdingImageUrl="https://foo/bar/baz.png"
      items={
        Array [
          Object {
            "duration": 100,
            "kind": "clip",
            "versionID": "bar",
          },
        ]
      }
      kind="clip"
      pid="foo"
      title="Hello World!"
    />
  </ForwardRef>
</React.Fragment>
`;

exports[`Video with no aresMedia should only render the video 1`] = `null`;

exports[`Video with no data should not render anything 1`] = `null`;
